space-concordia-robotics/robotics-prototype
# Power report 568

# Assignee Section

## Description
The power report node records power consumption starting when a start command is sent to it, and when stopped it saves it to a file. The latest power report is also retrievable from a service the node provides.

- Fixes on the pds node
- PDS simulator now sends more feeds
- Implemented PowerReportNode

### Steps for Testing
1. Ensure the rover is on this branch and started up (or use the local_rover.launch file in robot/util/rosRoverStart)
2. Run this command: ```rostopic pub /power_report_command std_msgs/String "data: 'start'"```. This will start the recording.
3. Wait some time
4. Run this command: ```rostopic pub /power_report_command std_msgs/String "data: 'stop'"```. This will stop the recording.
5. Check in ```~/Power_Reports``` for a power report.
6. Run this command: ```rosservice call /power_report_provider 0``` to see the latest report at any time

closes #568 

The approval from all software team leads is necessary before merging.

# Reviewer Section

Aside from local testing and the General Integration Test it is implied that static analysis should be included in the verification process.


For Pull Requests that do not include code changes, it is not required to perform the tests above.
